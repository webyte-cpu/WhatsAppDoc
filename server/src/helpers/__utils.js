import objectKeysToCamelCase from "./objectKeyCase.js";
import objectFilter from "./objectFilter.js";
import { constructTestServer, cleanDb ,} from "./integrationTesting.js";

export {
  constructTestServer,
  objectKeysToCamelCase,
  objectFilter,
  cleanDb,
};
