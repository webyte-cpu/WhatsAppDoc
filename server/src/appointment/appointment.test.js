import {
  GET_APPOINTMENT,
  CREATE_APPOINTMENT,
  UPDATE_APPOINTMENT,
  DELETE_APPOINTMENT
} from './queries.js';
import { constructTestServer } from '../helpers/__utils.js';
import { createTestClient } from 'apollo-server-testing';
import { query } from 'express';

describe("Queries and Mutations Testing", () => {
  
  const { server } = constructTestServer();
  
  const { query } = createTestClient(server);

  it("fetches an appointment", async () => {
 
    const res = await query({
      query: GET_APPOINTMENT,
      variables: {uid: }
    })
  })

  expect(res).toMatchSnapshot();
});

it("create an appointment", async () => {
  
  const res = await query({
    query: CREATE_APPOINTMENT,
    variables: {
      patientUid: ,
      doctorClinicUid: ,
      status: ,
      timestamp: ,
      doctorRemarks: ,
    }
  })

  expect(res).toMatchSnapshot();
})

it("updates an appointment", async () => {

  const res = await query({
    query: UPDATE_APPOINTMENT,
    variables: {
      patientUid: ,
      doctorClinicUid: ,
      status: ,
      timestamp: ,
      doctorRemarks: ,
    }
  })

  expect(res).toMatchSnapshot();
})

it("deletes a schedule", async () => {
  
  const res =  await query({
    query: DELETE_APPOINTMENT,
    variables: { uid: }
  })

  expect(res).toMatchSnapshot();
})

