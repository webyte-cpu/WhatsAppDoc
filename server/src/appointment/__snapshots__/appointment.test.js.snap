// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries and Mutations Testing create an appointment 1`] = `
Object {
  "data": Object {
    "createAppointment": Object {
      "createdAt": 2021-06-10T22:45:04.108Z,
      "dateTime": 2021-05-30T15:00:00.000Z,
      "doctorClinicUid": "6ecaae63-1196-4e48-818d-0266dbed39e2",
      "doctorRemarks": null,
      "patientUid": "7fc392c3-cd94-40ec-bd53-81de4aa8f8cf",
      "status": "PENDING",
      "uid": "b9040ec3-b3cc-4eaf-b501-084610549aae",
      "updatedAt": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries and Mutations Testing fetches all appointments 1`] = `
Object {
  "data": Object {
    "getAllAppointment": Array [
      Object {
        "createdAt": null,
        "dateTime": 2021-05-26T11:30:10.000Z,
        "doctorClinicUid": "740424a7-4e39-426f-aa53-65cf3e31609d",
        "doctorRemarks": "test remarks",
        "patientUid": "31e7e881-4b37-4cee-80f9-4ab32a25773d",
        "status": "PENDING",
        "uid": "1239e82b-b4ea-4e73-8861-df2ff69fac82",
        "updatedAt": null,
      },
      Object {
        "createdAt": null,
        "dateTime": 2021-11-21T18:00:00.000Z,
        "doctorClinicUid": "6ecaae63-1196-4e48-818d-0266dbed39e2",
        "doctorRemarks": "test remarks",
        "patientUid": "7fc392c3-cd94-40ec-bd53-81de4aa8f8cf",
        "status": "PENDING",
        "uid": "f621fddb-8b13-4c58-93d7-3159a99c06bc",
        "updatedAt": null,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries and Mutations Testing fetches an appointment 1`] = `
Object {
  "data": Object {
    "getAppointment": Array [
      Object {
        "createdAt": null,
        "dateTime": 2021-11-21T18:00:00.000Z,
        "doctorClinicUid": "6ecaae63-1196-4e48-818d-0266dbed39e2",
        "doctorRemarks": "test remarks",
        "patientUid": "7fc392c3-cd94-40ec-bd53-81de4aa8f8cf",
        "status": "PENDING",
        "uid": "f621fddb-8b13-4c58-93d7-3159a99c06bc",
        "updatedAt": null,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries and Mutations Testing updates an appointment 1`] = `
Object {
  "data": Object {
    "updateAppointment": Object {
      "dateTime": 2021-05-26T16:00:00.000Z,
      "doctorClinicUid": "6ecaae63-1196-4e48-818d-0266dbed39e2",
      "doctorRemarks": "test test test",
      "patientUid": "7fc392c3-cd94-40ec-bd53-81de4aa8f8cf",
      "status": "ON_GOING",
      "uid": "f621fddb-8b13-4c58-93d7-3159a99c06bc",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
